src/rnnodeapp/serve_styles.js

```diff
@@ -1,42 +1,86 @@
-const ecstatic = require('ecstatic')
-const url = require('url')
-const path = require('path')
-const fs = require('fs')
+const ecstatic = require('ecstatic');
+const url = require('url');
+const path = require('path');
+const fs = require('fs');
 
-module.exports = function createStyleServer (root) {
-  const styleFile = path.join(root, 'style.json')
+module.exports = function createStyleServer(root) {
+  const styleFile = path.join(root, '/style/style.json');
 
-  return function (req, res) {
+  const onError = err => {
+    console.error(`Error serving ${styleFile}:`, err);
+    res.statusCode = 404;
+    res.write(err);
+    res.end();
+  };
+
+  const serveStyleFile = (req, res) => {
+    fs.stat(styleFile, (err, stat) => {
+      if (err) return onError(err);
+
+      return fs.readFile(styleFile, 'utf8', (error, data) => {
+        if (err) return onError(error);
+
+        data = Buffer.from(
+          data.replace(/\{host\}/gm, `http://${req.headers.host}`)
+        );
+
+        res.setHeader('content-type', 'application/json; charset=utf-8');
+        res.setHeader('last-modified', new Date(stat.mtime).toUTCString());
+        res.setHeader('content-length', data.length);
+        res.setHeader(
+          'Access-Control-Allow-Headers',
+          'Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since'
+        );
+        res.setHeader('Access-Control-Allow-Origin', '*');
+        res.statusCode = 200;
+        res.write(data);
+        res.end();
+        return null;
+      });
+    });
+  };
+
+  const listFiles = (req, res) => {
+    const files = fs.readdirSync(root);
+    res.statusCode = 200;
+    const string = files.map(f => JSON.stringify(f)).join(',');
+    console.log(root, files, string);
+    res.write(string);
+    res.end();
+    return null;
+  };
+
+  const tiles = (req, res) => {
```

Comment:
Curious why this part is necessary and the ecstatic call at the bottom didn't work?